src/main/java/me/jellysquid/mods/lithium/mixin/entity/fast_piglin_brain/PiglinSpecificSensorMixin.java

```diff
@@ -0,0 +1,215 @@
+import net.minecraft.entity.mob.WitherSkeletonEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.predicate.entity.EntityPredicates;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.ai.brain.Brain;
+import net.minecraft.entity.ai.brain.MemoryModuleType;
+import net.minecraft.entity.ai.brain.sensor.PiglinSpecificSensor;
+
+@Mixin(PiglinSpecificSensor.class)
+public class PiglinSpecificSensorMixin {
+
+    /** A note to maintainers : 
+      * These id's are not safe between versions and most likely will change if a new mob is added,
+      * they are based on the order of entities being added to the ENTITY_TYPE registry
+      * You can check a value by getting the output of Registry.ENTITY_TYPE.get(int) 
+      * There is most likely a better way of doing this specifically however, static finals are inlined
+      * every time they are used and in a loop, inlining a check of a registry numeric id of a string isn't
+      * exactly amazing for performance.
+      * 
+      * If you are updating, check this code to make sure it is the expected entites being returned by those
+      * numeric id's.
+      * 
+      * If a new mob is added to this sensor then its ID should be added here and the accoring code added.
+    */
+    private static final int HOGLIN_ID = 33;
+    private static final int PIGLIN_ID = 60;
+    private static final int PLAYER_ID = 105;
+    private static final int ZOMBIFIED_ID = 104;
+
+
+    @Overwrite
+    protected void sense(ServerWorld world, LivingEntity entity) {
+        Brain<?> brain = entity.getBrain();
+        brain.remember(MemoryModuleType.NEAREST_REPELLENT, findSoulFire(world, entity));
+        Optional<MobEntity> nearestVisibleNemisis = Optional.empty();
+        Optional<HoglinEntity> nearestVisibleHuntableHoglin = Optional.empty();
+        Optional<HoglinEntity> nearestVisibleBabyHoglin = Optional.empty();
+        Optional<PiglinEntity> nearestVisibleBabyPiglin = Optional.empty();
+        Optional<LivingEntity> nearestVisibleZombified = Optional.empty();
+        Optional<PlayerEntity> nearPlayerNotInGold = Optional.empty();
+        Optional<PlayerEntity> playerHoldingWantedItem = Optional.empty();
+        List<PiglinEntity> visableAdultPiglins = Lists.newArrayList();
+        List<PiglinEntity> allAdultPiglins = Lists.newArrayList();
+
+        // This list appears to be sorted from nearest to furthest
+        List<LivingEntity> allVisibleMobs = (List)brain.getOptionalMemory(MemoryModuleType.VISIBLE_MOBS).orElse(ImmutableList.of());
+        Iterator visibleMobsIterator = allVisibleMobs.iterator();
+
+        Stream players = allVisibleMobs.stream().filter(EntityPredicates.EXCEPT_CREATIVE_SPECTATOR_OR_PEACEFUL);
+        Stream hoglins = allVisibleMobs.stream().filter((currentEntity) -> currentEntity.getEntityId() == HOGLIN_ID);
```

Comment:
I don't think `this.entityId = MAX_ENTITY_ID.incrementAndGet();` is what you want to use here. I also don't want these hardcoded IDs. I doubt the performance difference is so significant that we want this maintainance burden.